### PR TITLE
Blob storage edits

### PR DESCRIPTION
**Pull request by annikakouhia**
#### Description of changes

A few organizational/specification changes for the increase in download size using blobs that I merged last week. First, giving more meaningful variable names (in blob-provider and window-utils). Second, abstracting the creation of a blob and extraction of a URL from that blob into a class named file-url-provider.

Mocked the action of this class for all Export Dialog tests.

Removed WindowUtils from deps of Export Dialog since their utility was now handled by the File URL Provider.

#### Pull request checklist

- [X] Addresses an existing issue: Addition to Issue #730 (PR #766)
- [X] Added relevant unit test for your changes. (`yarn test`)
- [X] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [X] Ran precheckin (`yarn precheckin`)
- [N/A] (UI changes only) Added screenshots/GIFs to description above
- [N/A] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

File: src/tests/unit/tests/DetailsView/components/export-dialog.test.tsx [hunk 1]
```diff
@@ -52,35 +49,30 @@ describe('ExportDialog', () => {
...
         it.each(isOpenOptions)('with open %p', isOpen => {
-            provideBlobMock
-                .setup(p => p(It.isAny(), It.isAnyString()))
-                .returns(() => blobStub)
-                .verifiable(Times.once());
-            windowUtilsMock
-                .setup(w => w.createObjectURL(blobStub))
+            props.isOpen = isOpen;
+            fileProviderMock
+                .setup(f => f.provideURL(It.isAny(), It.isAnyString()))
```
Comment (Shobhit1) [inline src/tests/unit/tests/DetailsView/components/export-dialog.test.tsx]: nit: you can probably  use `blobStub` here to do a strick typecheck than `It.isAny()`

File: src/common/file-url-provider.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { provideBlob } from './blob-provider';
+import { WindowUtils } from './window-utils';
+
+export class FileURLProvider {
+    constructor(private windowUtils: WindowUtils) {}
```
Comment (Shobhit1) [inline src/common/file-url-provider.ts]: This needs tests?

Comment (smoralesd) [reply src/common/file-url-provider.ts]: Agreed... you'll need to pass `provideBlob` through the constructor if you want to mock it on the test too.

File: src/tests/unit/tests/DetailsView/components/export-dialog.test.tsx [hunk 3]
```diff
@@ -52,35 +49,30 @@ describe('ExportDialog', () => {
...
         it('closes the dialog onDismiss', () => {
             onCloseMock.setup(oc => oc()).verifiable(Times.once());
-            provideBlobMock
-                .setup(p => p(It.isAny(), It.isAnyString()))
-                .returns(() => blobStub)
-                .verifiable(Times.once());
-            windowUtilsMock
-                .setup(w => w.createObjectURL(blobStub))
+            fileProviderMock
+                .setup(f => f.provideURL(It.isAny(), It.isAnyString()))
```
Comment (smoralesd) [inline src/tests/unit/tests/DetailsView/components/export-dialog.test.tsx]: `f` inside the `setup` call is a little short for a name, how does `provider` sounds?

File: src/common/file-url-provider.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+    public provideURL = (blobParts?: any[], mimeType?: string): string => {
```
Comment (haonliu) [inline src/common/file-url-provider.ts]: Will this cause error if both params are not provided? i.e. calling provideBlob()

Comment (Shobhit1) [reply src/common/file-url-provider.ts]: This is a good idea for one of the test cases that @annikakouhia can add for this file.

Comment (haonliu) [reply src/common/file-url-provider.ts]: Also I think a test case should be added in blob provider.

Comment (annikakouhia) [reply src/common/file-url-provider.ts]: Made them both mandatory parameters--blobs can be created without both, but for our purposes we want both values

File: src/tests/unit/common/file-url-provider.test.tsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        const returnedURL = 'returned url';
+        const blobStub = {} as Blob;
+
+        provideBlobMock
+            .setup(pbm => pbm(content, mimeType))
+            .returns(() => blobStub)
+            .verifiable(Times.once());
+
+        windowUtilsMock
+            .setup(w => w.createObjectURL(blobStub))
```
Comment (Shobhit1) [inline src/tests/unit/common/file-url-provider.test.tsx]: nit: You can rename `w` inside the `setup` function to a better name. `windowsUtilProvider` maybe?

File: src/tests/unit/tests/DetailsView/components/export-dialog.test.tsx (hunk 1 above)
Comment (Shobhit1) [inline src/tests/unit/tests/DetailsView/components/export-dialog.test.tsx]: nit: same as above. `f` can be renamed to something better!

**Pull request merged**